nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -151,6 +152,35 @@ impl DataStore {
         .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
 
+    /// Fetch all disks (subject to pagparams)
+    pub async fn disk_list_all(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, String>,
+    ) -> ListResultVec<Disk> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+
+        use db::schema::disk::dsl;
+        paginated(dsl::disk, dsl::name, pagparams)
+            .filter(dsl::time_deleted.is_null())
+            .select(Disk::as_select())
+            .load_async::<Disk>(self.pool_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
+    /// Fetches the disk with the given UUID
+    pub async fn disk_get(&self, disk_uuid: Uuid) -> Result<Disk, Error> {
+        use db::schema::disk::dsl as disk_dsl;
+
+        disk_dsl::disk
+            .filter(disk_dsl::id.eq(disk_uuid))
+            .select(Disk::as_select())
+            .first_async::<Disk>(self.pool())
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
```

Comment:
Yeah, so the reason _not_ to do this is that it's exactly what `LookupPath` is _for_ except that LookupPath does an authz check.  I'm afraid by having this here, it would be very easy for people to use this function in Nexus and skip the authz check by accident, leaving a gaping security hole.

omdb's db.rs already imports a bunch of stuff to do db queries directly (e.g., for dns stuff).  Is it a lot more for disks?

If that's really a problem, we could modify `LookupPath` to add a `skip_authz()` option.  At least then people would have to go explicitly out of their way to skip authz.

Reply:
footgun removed from nexus, placed in omdb!